tests/data/sql-injection.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace RexSqlInjection;
+
+use rex;
+use rex_i18n;
+use rex_sql;
+
+/**
+ * @return void
+ */
+function safeArray($_id, string $langID)
+{
+    $select = rex_sql::factory();
+    $select->setTable('article');
+    $select->setWhere(['id' => $_id, 'clang_id' => $langID]);
+}
+
+/**
+ * @param numeric-string  $numericS
+ * @return void
+ */
+function safeScalars($mixed, string $s, $numericS, int $i, float $f, bool $b, array $arr)
+{
+    $select = rex_sql::factory();
+    $select->setTable('article');
+    $select->setWhere('id = ' . $select->escape($s));
+    $select->setWhere('id = ' . $select->escape($mixed));
+    $select->setWhere($select->escapeIdentifier($s). ' = ' . $select->escape($mixed));
+    $select->setWhere($select->escapeIdentifier($s). ' LIKE "' . $select->escapeLikeWildcards($mixed) . '"');
```

Comment:
der call ist aber nicht safe. also `escapeLikeWildcards` genügt nicht, da muss immer noch ein `escape` drum (oder halt als param in prepared query).
Oftmals will man nach `escapeLikeWildcards` ja noch tatsächliche Wildcards drum rum setzen.
Wenn man selbst escapen will, hätte man dann sowas:

```php
     $select->setWhere($select->escapeIdentifier($s). ' LIKE ' . $select->escape('%'.$select->escapeLikeWildcards($mixed).'%'));
```

Reply:
gutes auge. gefixed mit https://github.com/FriendsOfREDAXO/rexstan/pull/155/commits/e8d51f57d68d8550774e7325bab5d9a1fc7c536d